lib/neo4j/rspec/matchers/relations.rb

```diff
@@ -45,6 +45,30 @@ module Relations
             "expected the #{rel.class.name} relation not to have a relationship type #{type}"
           end
         end
+
+        matcher :use_create_unique do |arg|
+          match do |rel|
+            rel.class.unique? && (arg.nil? || arg == rel.class.creates_unique_option)
+          end
+
+          description do
+            msg = "use CREATE UNIQUE"
+            msg += " with option #{arg}" if arg
+            msg
+          end
+
+          failure_message do |rel|
+            msg = "expected the #{rel.class.name} relation to use CREATE UNIQUE clause"
+            msg += " with option #{arg}" if arg
+            msg
```

Comment:
If the `rel.class.creates_uniqe_option` is the wrong one (e.g., in the spec I ask for 3 keys, but only use 2 in my relationship), it would be useful to see them side-by-side in the message.  Perhaps with a message something like 

```ruby
msg = "expected the #{rel.class.name} relation to use CREATE UNIQUE clause"
msg += " with option #{arg}" if arg
msg += "   but found #{rel.class.creates_unique_option}
```